pkg/k8s/apis/cilium.io/v2alpha1/types.go

```diff
@@ -37,8 +37,10 @@ type CoreCiliumEndpoint struct {
 	// Encryption is the encryption configuration of the node
 
 	// +kubebuilder:validation:Optional
-	Encryption cilium_v2.EncryptionSpec `json:"encryption,omitempty"`
+	Encryption cilium_v2.EncryptionSpec `json:"encryption"`
 	NamedPorts models.NamedPorts        `json:"named-ports,omitempty"`
+	// ServiceAccount is the service account of the endpoint.
+	ServiceAccount string `json:"service-account,omitempty"`
```

Comment:
Incidentally, are there upgrade concerns we need to worry about here?

When initially rolling out the new Cilium version, presumably the new cilium-operator needs to be deployed first to update the CRD. Then individual agents should be able to populate the new field when updating the status of the CEPs.

Once the new version of Cilium is up and running, it will update all CEPs for the current node to include this information presumably (IIRC that's part of the per-endpoint controller to reconcile this info in k8s). Then after that, if CES is enabled, the operator should aggregate that new info and share it out via CES to all remote agents.

Then I guess if the cilium-agent is downgraded, it should just silently ignore the extra field?